### DISCUSSION TITLE
How to access devices in robots on WeBot

### DISCUSSION THREAD

**Discussion by Alanl101** [cyberbotics/webots#6356] [General]
I want to be able to see how to see different devices on my e-puck robot. I want to see how many sensors and cameras I have. Currently I've tried looking in the file tree and trying to print out some code on python but no luck.

**Comment by Alanl101** [comment]
Duplicate

**Comment by Qtbgl** [comment]
I tried to print all the devices on NAO, and here is the code :
```python
    n = robot.getNumberOfDevices()
    for i in range(n):
        device = robot.getDeviceByIndex(i)
        print(device.getName())
```
